tasks/patch_wordpress.js

```diff
@@ -280,4 +285,76 @@ module.exports = function(grunt) {
 
 	})
 
+	grunt.registerTask( 'upload_patch_to_wordpress_trac', 'Upload the current diff of your develop-wordpress directory to WordPress trac', function( ticketNumber ) {
+		var done = this.async()
+		var options = this.options(defaults)
+
+		grunt.log.debug( 'ticketNumber: ' + ticketNumber )
+		grunt.log.debug( 'options: ' + JSON.stringify( options ) )
+
+		var requestTicketNumber = function() {
+			rl.question('Enter the ticket number: ', function(_ticketNumber) {
+				_ticketNumber = parseInt( _ticketNumber, 10 )
+				if ( typeof _ticketNumber != 'number' ) {
+					grunt.log.error( '"' + _ticketNumber + '" is not a valid ticketNumber.' )
+					requestTicketNumber()
+				} else {
+					ticketNumber = _ticketNumber
+					requestCredentials()
+				}
+			})
+		}
+
+		var requestCredentials = function() {
+			rl.question('Enter your wordpress.org username: ', function(username) {
+				rl.question('Enter your wordpress.org password: ', function(password) {
+					uploadPatchWithCredentials( username, password )
+				})
+			})
+		}
+
+		var uploadPatchWithCredentials = function( username, password ) {
+			var diffCommand = is_svn() ? 'svn diff' : 'git diff'
+
+			exec( diffCommand, function(error, result, code) {
+				var client = xmlrpc.createSecureClient({
+					hostname: options.tracUrl,
+					port: 443,
+					path: '/login/xmlrpc',
+					basic_auth: {
+						user: username,
+						pass: password
+					}
+				})
+				client.methodCall(
+					'ticket.putAttachment',
+					[
+						ticketNumber,
+						ticketNumber + '.diff',
+						'', // description. empty for now.
+						new Buffer( new Buffer(result).toString('base64'), 'base64'),
+						false // never overwrite the old file
+					], function( err, value ) {
+						if ( err === null ) {
+							grunt.log.writeln( 'Uploaded patch.' )
+							process.exit()
+						} else {
+							grunt.log.error( 'Something went wrong: ' )
+							grunt.log.error( err )
+							grunt.log.error( 'Let\'s start over.' )
```

Comment:
I like failing and erroring out rather than starting over and forcing someone to quit.  That may just be personal preference, I don't know.